docs/source/rbac.rst

```diff
@@ -292,108 +295,197 @@ For example:
 
 .. _rbac-using_rbac:
 
+Automatically granting roles based on the user LDAP group membership
+--------------------------------------------------------------------
+
+.. note::
+
+   This functionality is only available in |st2| v2.3.0 and above when LDAP auth backend is used
+   for authentication.
+
+In addition to manually assigning roles to the users based on the definitions in the
+``/opt/stackstorm/rbac/assignments/`` directory, |st2| also supports automatically granting roles
+to users on authentication based on the LDAP groups membership.
+
+This comes handy in enterprise environments and makes |st2| user provisioning easier and faster.
+It means operator doesn't need to manually write and manage RBAC role assignment files on disk,
+because roles are automatically granted to the users based on their LDAP group membership and
+mappings files in ``/opt/stackstorm/rbac/mappings/`` directory.
+
+To be able to utilize this feature it first needs to be enabled in ``st2.conf`` by setting
+``rbac.sync_remote_groups`` option to ``True``.
+
+.. code-block:: ini
+
+  [rbac]
+  sync_remote_groups = True
+
+After this feature is enabled, operator needs to write mapping files which tell |st2| which roles
+to automatically grant to users which are a member of a particular LDAP group. Mapping files are
+located in ``/opt/stackstorm/mappings/`` directory and map LDAP group to one or more |st2| roles.
+
+Two examples of such mapping files can be found below.
+
+.. note::
+
+  LDAP group names referenced in ``group`` attribute are case-sensitive.
+
+``/opt/stackstorm/rbac/mappings/stormers.yaml``
+
+.. code-block:: yaml
+
+    ---
+      group: "CN=stormers,OU=groups,DC=stackstorm,DC=net"
+      description: "Automatically grant admin role to all stormers group members."
+      roles:
+        - "admin"
+
+Each user which is a member of ``CN=stormers,OU=groups,DC=stackstorm,DC=net`` LDAP group will
+automatically be granted ``admin`` |st2| role when they successfully authenticate with |st2|.
+
+``/opt/stackstorm/rbac/mappings/testers.yaml``
+
+.. code-block:: yaml
+
+    ---
+      group: "CN=testers,OU=groups,DC=stackstorm,DC=net"
+      description: "Automatically grant observer and q_admin role to all testers group members."
+      roles:
+        - "observer"
+        - "qa_admin"
+
+Each user which is a member of ``CN=testers,OU=groups,DC=stackstorm,DC=net`` LDAP group will
+automatically be granted ``observer`` and ``qa_admin`` |st2| role when they successfully
+authenticate with |st2|.
+
+Once the mapping definitions files are written, operator needs to run
+``st2-apply-rbac-definitions`` tool to store those definitions in the database. This tool also
+needs to be ran after any change or removal of mappings files.
+
+How it works
```

Comment:
Since you fixed up some other heading capitalization (thanks!!) should we also do it here?